### PR TITLE
Add zksync mainnet & testnet support

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
Fixes: #2309 

Add zksync mainnet & testnet support

## Checklist

This PR:

- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't disable eslint rules.
- [x] Doesn't work around the type checker (including but not limited to: type casts, non-null assertions, `@ts-ignore`, unjustified optional values).
- [x] Doesn't contain commented out code.
- [x] If adding/updating a feature, it adds/updates its test script on Notion.

### CODE REVIEW

**Review by melissa-neira — APPROVED** [review]
lgtm!

**Review by michellema1208 — DISMISSED** [review]
Screen Shot 2023-12-12 at 3 47 56 PM

@bhargavaparoksham @gravityblast - do we have Lit support on zkSync yet? Looks like my application to a zkSync round is failing on encryption step

File: packages/builder/src/utils/wallet.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { ethers } from "ethers";
 import PGNIcon from "common/src/icons/PublicGoodsNetwork.svg";
 import AVAXIcon from "common/src/icons/AVAX.svg";
 import POLIcon from "common/src/icons/pol-logo.svg";
+import ZKSYNCICON from "common/src/icons/zksync-logo.svg";
```
Comment (bard) [inline packages/builder/src/utils/wallet.ts:8]: Looks to be accidentally all uppercase.

File: packages/grant-explorer/src/features/api/utils.ts [hunk 2]
```diff
@@ -228,6 +242,92 @@ const FANTOM_TESTNET_TOKENS: VotingToken[] = [
...
+    logo: TokenNamesAndLogos["USDT"],
+    redstoneTokenId: RedstoneTokenIds["USDT"],
+    defaultForVoting: false,
+    canVote: true,
+  },
+  {
+    name: "LUSD",
+    chainId: ChainId.ZKSYNC_ERA_MAINNET_CHAIN_ID,
+    address: "0x503234F203fC7Eb888EEC8513210612a43Cf6115",
+    decimal: 18,
```
Comment (bard) [inline packages/grant-explorer/src/features/api/utils.ts:313]: Should this be 6 or 18?

File: packages/round-manager/src/features/api/utils.ts [hunk 3]
```diff
@@ -222,6 +236,76 @@ const FANTOM_TESTNET_TOKENS: PayoutToken[] = [
...
+    address: "0x493257fD37EDB34451f62EDf8D2a0C418852bA4C",
+    decimal: 6,
+    logo: TokenNamesAndLogos["USDT"],
+    redstoneTokenId: RedstoneTokenIds["USDT"],
+  },
+  {
+    name: "LUSD",
+    chainId: ChainId.ZKSYNC_ERA_MAINNET_CHAIN_ID,
+    address: "0x503234F203fC7Eb888EEC8513210612a43Cf6115",
+    decimal: 18,
```
Comment (bard) [inline packages/round-manager/src/features/api/utils.ts:295]: (Same question as above.)

File: packages/grant-explorer/src/features/api/utils.ts (hunk 2 above)
Comment (vacekj) [reply packages/grant-explorer/src/features/api/utils.ts:313]: The contract says 18, so I would say 18.

**Review by bard — APPROVED** [review]

**Pull request merged**